src/tools/tidy/src/ui_tests.rs

```diff
@@ -128,4 +132,44 @@ pub fn check(path: &Path, bad: &mut bool) {
             }
         }
     });
+
+    // Check that any test files modified don't have a useless name
+    for changed_file in git::get_git_modified_files(None, &vec!["rs"]).unwrap().unwrap() {
+        // `get_git_modified_files` reports all files in the repo, only check ui tests
+        if !paths.iter().any(|path| changed_file.starts_with(&path.to_string_lossy().into_owned()))
+        {
+            continue;
+        }
+
+        let changed_file = changed_file.as_str();
+
+        // use git rev-list to make sure that the file still exists, since `get_git_modified_files`
+        // names both the source and destination files in a rename.
+        let mut cmd = Command::new("git");
+        cmd.args(["rev-list", "-1", "--all"]);
+        cmd.arg(changed_file);
+
+        // If a non-zero status was returned, it's from a deleted file, otherwise it exists
+        let is_modified = cmd.output().expect("unable to execute `git rev-list`").status.success();
+
+        if is_modified {
+            lazy_static! {
+                static ref ISSUE_NAME_REGEX: Regex = Regex::new(r"issues?[-_]?\d+").unwrap();
+                static ref ICE_NAME_REGEX: Regex = Regex::new(r"ice[-_]?\d+").unwrap();
+            }
+
+            let name =
+                Path::new(changed_file).file_stem().and_then(OsStr::to_str).expect("non-UTF8 path");
+            for regex in [&*ISSUE_NAME_REGEX, &*ICE_NAME_REGEX] {
+                if let Some(m) = regex.find(name) {
+                    tidy_error!(
+                        bad,
+                        "test file {} should not use {} in its name. link the issue in a comment in the test and use a more descriptive name for the test instead.",
```

Comment:
```suggestion
                        "test file `{}` should not use `{}` in its name. link the issue in a comment in the test and use a more descriptive name for the test instead.",
```

Reply:
Can this be slightly clearer that it isn't the issue number that is totally forbidden from the filename, but that the filename needs to actually identify what the test is about at-a-glance, i.e. `issue-\d+-what-it-tests` is acceptable?